x/cronos/migrations/v2/migrate_test.go

```diff
@@ -0,0 +1,47 @@
+package v2_test
+
+import (
+	"testing"
+
+	"github.com/crypto-org-chain/cronos/x/cronos/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/crypto-org-chain/cronos/app"
+)
+
+func TestMigrationTestSuite(t *testing.T) {
+	suite.Run(t, new(MigrationTestSuite))
+}
+
+type MigrationTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *app.App
+
+	params types.Params
+}
+
+func (suite *MigrationTestSuite) SetupTest() {
+	// account key
+	priv, err := ethsecp256k1.GenerateKey()
+	require.NoError(suite.T(), err)
+	address := common.BytesToAddress(priv.PubKey().Address().Bytes())
+	cronosAdmin := sdk.AccAddress(address.Bytes()).String()
+
+	suite.app = app.Setup(suite.T(), cronosAdmin, true)
+	suite.ctx = suite.app.NewContext(false, tmproto.Header{})
+	suite.params = types.DefaultParams()
+	suite.params.CronosAdmin = cronosAdmin
+	suite.params.EnableAutoDeployment = true
+}
+
+func (suite *MigrationTestSuite) TestMigrate() {
+	suite.Require().Equal(suite.params, suite.app.CronosKeeper.GetParams(suite.ctx))
```

Comment:
Will we test real migration here?

Reply:
yeah, this should not call `migrate` here. I need to think about how to test real migration

Reply:
testing real migration requires either two binaries with different versions or a before-upgrade database, which seems not easy to be done. Do you have better ideas?

Reply:
I think a [test](https://github.com/adu-crypto/cronos/pull/6/commits/b7dd95e8434fb73ee5fc29df719e746f111fb858) for migrate with store check is enough.